### PR TITLE
ldap: Add user DN attributes list config param

### PR DESCRIPTION
**Pull request by donatello**
## Description

This change uses the updated ldap library in minio/pkg (which was bumped
up to v3). New config parameter is added for LDAP configuration to
specify extra user attributes to load from the LDAP server and to store
it as additional claims for the user.

A test is added in sts_handlers.go that shows how to access the LDAP
attributes as a claim.

This is in preparation to add SSH pubkey authentication to MinIO's SFTP
integration.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,6 +2,12 @@ module github.com/minio/minio
 
 go 1.21
 
+replace github.com/minio/pkg/v3 => github.com/donatello/pkg/v3 v3.0.0-20240516203215-ffb947cd8ecd
```
Comment (zveinn) [inline go.mod]: Don't forget to replace these :)

**Review by zveinn — APPROVED** [review]

**Review by taran-p — APPROVED** [review]
Looks good once console is updated

**Comment by harshavardhana** [comment]
Waiting to be merged with relevant go.mod changes.

**Comment by donatello** [comment]
Updated PR with tag for minio/pkg/v3 but still needs console update + release.

File: go.mod [hunk 2]
```diff
@@ -2,6 +2,8 @@ module github.com/minio/minio
 
 go 1.21
 
+replace github.com/minio/console => github.com/donatello/console v0.12.6-0.20240522161239-e2303ef9d681
```
Comment (harshavardhana) [inline go.mod]:
```suggestion
replace github.com/minio/console => github.com/donatello/console v0.12.6-0.20240522161239-e2303ef9d681
```

You can point this to console master branch now.

File: go.mod [hunk 3]
```diff
@@ -55,7 +55,8 @@ require (
 	github.com/minio/madmin-go/v3 v3.0.52
 	github.com/minio/minio-go/v7 v7.0.70
 	github.com/minio/mux v1.9.0
-	github.com/minio/pkg/v2 v2.0.19
+	github.com/minio/pkg/v2 v2.0.17
```
Comment (harshavardhana) [inline go.mod]: ?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: go.mod (hunk 3 above)
Comment (donatello) [reply go.mod]: Fixed now, I think it was possibly because of changes since the PR was sent.

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**